packages/primitives/test/emotion-primitives.test.tsx

```diff
@@ -16,14 +16,15 @@ describe('Emotion primitives', () => {
   })
 
   test('should throw an error when used invalid primitive', () => {
+    // @ts-expect-error
     expect(() => styled.TEXT({})).toThrow()
   })
 
   test('should render the primitive when styles applied using object style notation', () => {
     const Text = styled.Text`
       color: red;
       font-size: 20px;
-      background-color: ${props => props.back};
+      background-color: ${(props: any) => props.back};
```

Comment:
could we declare those props using generics rather than using `any`?

Reply:
`styled.Text` does not currently take any generic parameters. Are you suggesting we change the types so that it behaves similarly to `styled` from `@emotion/styled`? Meaning that on this line of code, TypeScript would know that `props` has type `TextProps`.

I believe that goes beyond just converting the package to TypeScript since it would mean changing user-facing types, but we can do that if you want to.

Reply:
> styled.Text does not currently take any generic parameters. Are you suggesting we change the types so that it behaves similarly to styled from @emotion/styled? Meaning that on this line of code, TypeScript would know that props has type TextProps.

That would be ideal. It might not be the easiest task though - so let me know if you prefer not to do it right now.

> I believe that goes beyond just converting the package to TypeScript since it would mean changing user-facing types, but we can do that if you want to.

Are there any user-facing types here though? I can't find any.